Moodis/Moodis/Feature/Group/GroupViewModel.cs

```diff
@@ -0,0 +1,245 @@
+        {
+            group = new Group(name, SignInViewModel.currentUser.username);
+
+            var groupList = new List<Group>();
+            var userList = new List<User>();
+
+            if (File.Exists(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData) + "/groups.bin"))
+            {
+                groupList = Serializer.Load<List<Group>>(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData) + "/groups.bin");
+            }
+
+            if (!groupList.Exists(x => x.name == name && x.manager == SignInViewModel.currentUser.username))
+            {
+                groupList.Add(group);
+                Serializer.Save(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData) + "/groups.bin", groupList);
+            }
+
+            if (File.Exists(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData) + "/users.bin"))
+            {
+                userList = Serializer.Load<List<User>>(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData) + "/users.bin");
+            }
+
+            if (userList.Exists(user => user.username == SignInViewModel.currentUser.username))
+            {
+                int index = userList.FindIndex(user => user.username == SignInViewModel.currentUser.username);
+                userList[index].groupName = name;
+                SignInViewModel.currentUser.groupName = name;
+                Serializer.Save(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData) + "/users.bin",userList);
+            }
+            return group;
+        }
+
+        public Group getGroup()
+        {
+            var groupList = new List<Group>();
+
+            if (File.Exists(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData) + "/groups.bin"))
+            {
+                groupList = Serializer.Load<List<Group>>(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData) + "/groups.bin");
+            }
+
+            if (groupList.Exists(x => x.manager == SignInViewModel.currentUser.username))
+            {
+                group = groupList.Find(x => x.manager == SignInViewModel.currentUser.username);
+                return group;
+            }
+            else
+            {
+                if (groupList.Exists(x => x.name == SignInViewModel.currentUser.groupName))
+                {
+                    group = groupList.Find(x => x.name == SignInViewModel.currentUser.groupName);
+                    return group;
+                }
+                else
+                {
+                    group = null;
+                    return null;
+                }
+            }
+        }
+        public Group joinGroup(string groupName)
+        {
+            var groupList = new List<Group>();
+            var userList = new List<User>();
+
+            if (File.Exists(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData) + "/groups.bin"))
```

Comment:
This if block is used in many places, maybe it would be better to extract it to a private method called `fetchGroupList`?

Reply:
Every block does a different thing, so there is no use to extract it to a private method